### ISSUE TITLE
Arrays in plugin configuration

### ISSUE THREAD

**Issue by volodymyrpavlenko** [kevholditch/terraform-provider-kong#11]
Hi,
I ran into an issue when configuring kong with your provider. I'm trying to configure request-transformer plugin and I cannot use the array notation of the configuration. I cannot use comma-separated values, since my Header value contains comma, and kong doc says:

> Note: if the value contains a , then the comma separated format cannot be used. The array notation must be used instead.

So, here's what I want to achieve, note that config.add.headers is an array:
```
curl -X POST http://localhost:8001/apis/mockbin/plugins \
  --header 'content-type: application/json' \
  --data '{"name": "request-transformer", "config": {"add": {"headers": ["h1:v2", "h2:v1"]}}}'
```

I could not find a way to do this with your provider, could you add possibility to configure arrays?

Plugin documentation can be found here: https://getkong.org/plugins/request-transformer/?_ga=2.107772418.1474587009.1519637315-1381908740.1518784690

**Comment by kevholditch** [comment]
Can you please provide an example of the `kong_plugin` resource that you tried in HCL

**Comment by volodymyrpavlenko** [comment]
Thanks for a quick answer, here it goes:
```
resource "kong_plugin" "test_request_transformer" {
  api_id = "${kong_api.api.id}"
  name = "request-transformer"

  config = {
    add.headers ="X-Header1:value1,value2,X-Header2:value2,value3"
  }
}
```
Request to kong fails, resulting in the following error:

> error updating kong plugin: could not update plugin, error: {"config.add.headers":"key 'value2' has no value"}

**Comment by kevholditch** [comment]
I think your config is incorrect, the way that plugin seems to work is you define a header then you give it a value.  You are defining a header called `value2` but not giving it a value as the kong error says.

The following configuration works:

```
resource "kong_plugin" "test_request_transformer" {
  api_id = "${kong_api.api.id}"
  name = "request-transformer"
 
  config = {
    add.headers = "X-Header1:value1,X-Header2:value2"
  }
}
```

**Comment by volodymyrpavlenko** [comment]
You're right, this is how kong understands my request. But what I want to do is to send a header value that contains comma. Here's working Kong HTTP request, which I'm struggling to implement in terraform.
```
POST /apis/<API-UUID>/plugins/
{
	"name": "request-transformer",
	"config": {
		"add": {
			"headers": [
				"X-Header1:value1,value2",
				"X-Header2:value2,value3"
			]
		}
	}
}
```

**Comment by kevholditch** [comment]
I see now, I think we may have to change the provider to support key value pairs as well as a map.  Similar to how the `kong_consumer_plugin_config` works, notice how it has two config files `config_json` and `config`.  I think we may need to do something similar to pass through the values as are.  Do you want to create a branch and have a go?  Start with a failing test

**Comment by volodymyrpavlenko** [comment]
Yes, that would work! I'll try to prepare a PR, was longing to try go :)

**Comment by kevholditch** [comment]
You may want to start by getting this working in [gokong](https://github.com/kevholditch/gokong) which is the client that this terraform provider uses under the bonnet to talk to Kong.  I would suggest creating a test inside the `plugins_test.go` file in gokong, get it working in there and then you'll know how you need to expose it out in terraform.  I think you may well need to change `gokong` too.

**Comment by jmakanjuola** [comment]
Hi Kev,

Do I need to create an ACL group separately before defining the acl group within the kong_consumer_plugin_config?

**Comment by jmakanjuola** [comment]
Figured it out. Thanks.
